# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerManager.h
# Commit: dabd7d6836c8
# Full Hash: dabd7d6836c8ed368c06537c3c19a61295d67a18
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-21 04:13:05
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 4: Make ContentParent KeepAlives explicit with RAII references, r=smaug,dom-worker-reviewers,asuth
#   
#   This is a fairly significant patch, however it would be difficult to break it
#   down into smaller patches:
#   
# ==============================================================================

diff -r 9e04f49c926e -r dabd7d6836c8 dom/workers/remoteworkers/RemoteWorkerManager.h
--- a/dom/workers/remoteworkers/RemoteWorkerManager.h	Thu Jun 20 19:24:51 2024 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerManager.h	Thu Jun 20 19:24:51 2024 +0000
@@ -61,19 +61,24 @@
   RemoteWorkerManager();
   ~RemoteWorkerManager();
 
-  RemoteWorkerServiceParent* SelectTargetActor(const RemoteWorkerData& aData,
-                                               base::ProcessId aProcessId);
+  struct TargetActorAndKeepAlive {
+    RefPtr<RemoteWorkerServiceParent> mActor;
+    UniqueThreadsafeContentParentKeepAlive mKeepAlive;
+  };
 
-  RemoteWorkerServiceParent* SelectTargetActorInternal(
+  TargetActorAndKeepAlive SelectTargetActor(const RemoteWorkerData& aData,
+                                            base::ProcessId aProcessId);
+
+  TargetActorAndKeepAlive SelectTargetActorInternal(
       const RemoteWorkerData& aData, base::ProcessId aProcessId) const;
 
   void LaunchInternal(RemoteWorkerController* aController,
                       RemoteWorkerServiceParent* aTargetActor,
-                      const RemoteWorkerData& aData,
-                      bool aRemoteWorkerAlreadyRegistered = false);
+                      UniqueThreadsafeContentParentKeepAlive aKeepAlive,
+                      const RemoteWorkerData& aData);
 
   using LaunchProcessPromise =
-      MozPromise<RefPtr<RemoteWorkerServiceParent>, nsresult, true>;
+      MozPromise<TargetActorAndKeepAlive, nsresult, true>;
   RefPtr<LaunchProcessPromise> LaunchNewContentProcess(
       const RemoteWorkerData& aData);
 